### PR TITLE
DAML profiler: Use non-blocking IO for writing profiles

### PR DESCRIPTION
**Pull request by hurryabit**
As suggested by @SamirTalwar-DA.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/6184)

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala [hunk 1]
```diff
@@ -92,9 +91,7 @@ object Profile {
     ) {
       def write(path: Path): Unit = {
         import JsonProtocol.fileFormat
-        val writer = new BufferedWriter(new FileWriter(path.toFile()))
-        writer.write(this.toJson.compactPrint)
-        writer.close()
+        val _ = Files.write(path, this.toJson.compactPrint.getBytes)
```
Comment (hurryabit) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala:94]: @S11001001 Does `spray-json` provide a way to write the JSON object into a file without first converting it to a string? We're talking about fairly big JSON objects of multiple MB here.

Comment (SamirTalwar-DA) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala:94]: `getBytes` clones the string. I'd suggest using `Seq(this.toJson.compactPrint).asJava` instead.

Comment (SamirTalwar-DA) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala:94]: I had a quick look into it and couldn't find anything that returned a `Writer` or `OutputStream`, unfortunately.

Comment (hurryabit) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Profile.scala]: I wasn't aware of that. Thanks for pointing it out. I changed it accordingly.

**Comment by hurryabit** [comment]
This is part of https://github.com/digital-asset/daml/pull/5957.

**Pull request merged**